### PR TITLE
Feat/cairo 2.3.0 rc0

### PR DESCRIPTION
**Pull request by Th0rgal**
This pull request adds both verification from Cairo 1 verifier (but following new identity interface) and a test.

### CODE REVIEW

File: src/lib.cairo [hunk 1]
```diff
@@ -59,7 +64,40 @@ mod Verifier {
             field: felt252,
             data: felt252,
             sig: (felt252, felt252)
-        ) {}
+        ) {
+            let caller = get_caller_address();
+            let starknetid_contract = self.starknetid_contract.read();
+            // todo: commented this 2 lines to run tests as we cannot mock calls w/ protostar
```
Comment (irisdv) [inline src/lib.cairo]: we can remove this comment

File: src/lib.cairo [hunk 2]
```diff
@@ -16,7 +19,9 @@ trait IVerifier<TContractState> {
 trait IStarknetID<TContractState> {
     fn owner_of(self: @TContractState, token_id: felt252) -> starknet::ContractAddress;
 
-    fn set_verifier_data(self: @TContractState, token_id: felt252, field: felt252, data: felt252);
+    fn set_verifier_data(
+        self: @TContractState, token_id: felt252, field: felt252, data: felt252, domain: u32
```
Comment (irisdv) [inline src/lib.cairo]: shouldn't we use `u128` for `token_id` to have the same type as in the identity contract ?

File: src/lib.cairo [hunk 3]
```diff
@@ -59,7 +64,40 @@ mod Verifier {
...
+            let starknetid_contract = self.starknetid_contract.read();
+            // todo: commented this 2 lines to run tests as we cannot mock calls w/ protostar
+            let owner = super::IStarknetIDDispatcher { contract_address: starknetid_contract }
+                .owner_of(token_id);
+            assert(caller == owner, 'Caller is not owner');
+
+            // ensure confirmation is not expired
+            let current_timestamp = get_block_timestamp();
+            assert(
+                current_timestamp <= timestamp.try_into().expect('error converting felt to u64'),
```
Comment (irisdv) [inline src/lib.cairo]: we could use timestamp as u64 so we don't have to convert it in this assert

**Review by irisdv — APPROVED** [review]
lgtm! I just left a few comments about typing in the code

File: src/lib.cairo (hunk 2 above)
Comment (Th0rgal) [reply src/lib.cairo]: right

**Pull request merged**